src/consumer/consumer.js

```diff
@@ -387,6 +387,37 @@ export default class Consumer {
     });
   }
 
+  /**
+   * Before writing the files into the file-system, remove the path-prefix that is shared among the main component files
+   * and its dependencies. It helps to avoid large file-system paths.
+   *
+   * This is relevant for IMPORTED components only when the author may have long paths that are not needed for whoever
+   * imports it. NESTED and AUTHORED components are written as is.
+   */
+  removeOriginallySharedDir(componentsAndDependenciesArray: ComponentWithDependencies[]): void {
+    const pathWithoutSharedDir = (pathStr, sharedDirToRemove) => {
+      return pathStr.replace(sharedDirToRemove + path.sep, '');
+    };
+    componentsAndDependenciesArray.forEach((componentAndDependencies) => {
+      componentAndDependencies.component.setOriginallySharedDir();
+      if (componentAndDependencies.component.originallySharedDir) {
+        const sharedDir = componentAndDependencies.component.originallySharedDir;
+        componentAndDependencies.component.files.forEach((file) => {
+          file.path = pathWithoutSharedDir(file.path, sharedDir);
+        });
+        componentAndDependencies.component.mainFile = pathWithoutSharedDir(
```

Comment:
The main file is always saved with / (in windows as well, so it might not work). please validate it on windows.